src/Nri/Ui/SegmentedControl/V10.elm

```diff
@@ -0,0 +1,305 @@
+module Nri.Ui.SegmentedControl.V10 exposing
+    ( Option, view
+    , SelectOption, viewSelect
+    , Width(..)
+    )
+
+{-| Changes from V9:
+
+  - hides non-displayed content rather than fully removing from the DOM, allowing for the segmented control to be used as a sticky-header that doesn't lose scroll position
+  - :skull: removes NavConfig and SelectConfig
+  - combines `view` and `viewSpa` (for V9 `view` behavior, be sure `toUrl` is Nothing. for V9 `viewSpa` behavior, pass through a Just as `toUrl`)
+  - add custom attributes hole to the Option (in order to make SegmentedControls compatible with the Modal component)
+  - combine `css` attributes into one to prevent class-name-order-change css :bug:s
+  - :bug: fix overflowing-y svg icon issue
+
+@docs Option, view
+@docs SelectOption, viewSelect
+@docs Width
+
+-}
+
+import Accessibility.Styled exposing (..)
+import Accessibility.Styled.Aria as Aria
+import Accessibility.Styled.Role as Role
+import Accessibility.Styled.Widget as Widget
+import Css exposing (..)
+import EventExtras
+import Html.Styled
+import Html.Styled.Attributes as Attributes exposing (css, href)
+import Html.Styled.Events as Events
+import Html.Styled.Keyed as Keyed
+import Nri.Ui
+import Nri.Ui.Colors.Extra exposing (withAlpha)
+import Nri.Ui.Colors.V1 as Colors
+import Nri.Ui.Fonts.V1 as Fonts
+import Nri.Ui.Html.Attributes.V2 as AttributesExtra
+import Nri.Ui.Svg.V1 as Svg exposing (Svg)
+import Nri.Ui.Util exposing (dashify)
+
+
+{-| -}
+type Width
+    = FitContent
+    | FillContainer
+
+
+{-| -}
+type alias SelectOption value msg =
+    { value : value
+    , label : String
+    , attributes : List (Attribute msg)
+    , icon : Maybe Svg
+    }
```

Comment:
how would you feel about inlining this for better error messages? It'd make the signature of `viewSelect` pretty complex (but maybe being explicit about the complexity is fine?)

Reply:
I somewhat believe that users may want to use the `SelectOption` and `Option` types. We're using it in the Example currently.. and for the Tab component, I definitely see people using the `Tab` type.

Reply:
seems fine!